go/ct/spc/specification.go

```diff
@@ -719,6 +719,129 @@ func getAllRules() []Rule {
 		effect:    NoEffect().Apply,
 	})...)
 
+	// --- TLOAD ---
+
+	rules = append(rules, rulesFor(instruction{
+		op:        TLOAD,
+		staticGas: 100,
+		pops:      1,
+		pushes:    1,
+		parameters: []Parameter{
+			StorageAccessKeyParameter{},
+		},
+		conditions: []Condition{
+			RevisionBounds(R13_Cancun, NewestSupportedRevision),
+			IsTransientSet(Param(0)),
+		},
+		effect: func(s *st.State) {
+			key := s.Stack.Pop()
+			value := s.Transient.GetStorage(key)
+			s.Stack.Push(value)
+		},
+	})...)
+
+	rules = append(rules, rulesFor(instruction{
+		op:        TLOAD,
+		staticGas: 100,
+		pops:      1,
+		pushes:    1,
+		parameters: []Parameter{
+			StorageAccessKeyParameter{},
+		},
+		conditions: []Condition{
+			RevisionBounds(R13_Cancun, NewestSupportedRevision),
+			IsTransientNotSet(Param(0)),
+		},
+		effect: func(s *st.State) {
+			key := s.Stack.Pop()
+			value := s.Transient.GetStorage(key)
+			s.Stack.Push(value)
+		},
+	})...)
+
+	rules = append(rules, Rule{
+		Name: "tload_pre_cancun",
+		Condition: And(
+			RevisionBounds(R07_Istanbul, R11_Paris),
+			Eq(Status(), st.Running),
+			Eq(Op(Pc()), TLOAD),
+			Ge(Gas(), 100),
```

Comment:
gas, and stacksize should not be conditions here, since it should fail regardless of those values.

Reply:
that is exactly my point, if you want it to fail explicitly because of the revision, you do not care about gas or stacksize.